### PR TITLE
Unresolve matches when there are new duplicates

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

If a resolved match:
"2 candidates with postcode W6 BHT and DOB 04/06/1989"

now has 3 candidates, when a new candidate is added to the match,
we should mark this match as unresolved again.

## Guidance to review

1. Enable feature duplicate_matching
2. Run the worker `UpdateFraudMatchesWorker.perform_async`
3. Then mark one match as resolved
4. Add one more duplicate candidate to this resolved match
5. Run the worker `UpdateFraudMatchesWorker.perform_async`
3. Verify the fraud_matches is in the database unresolved (or in the support UI)

## Link to Trello card

https://trello.com/c/wJRKYuMn/4383-new-candidate-duplicates-should-be-mark-unresolved

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/services/update_duplicate_matches_spec.rb [hunk 1]
```diff
@@ -39,6 +39,22 @@
           expect(candidate2.reload.submission_blocked).to be(false)
         end
       end
+
+      context 'when the candidates have been mannually resolved and new candidate is added to the match' do
```
Comment (avinhurry) [inline spec/services/update_duplicate_matches_spec.rb:43]:
```suggestion
      context 'when a duplicate match has been manually resolved and a new candidate is added to the match' do
```

File: spec/services/update_duplicate_matches_spec.rb [hunk 2]
```diff
@@ -39,6 +39,22 @@
...
+        let(:candidate3) { create(:candidate, email_address: 'exemplar3@example.com') }
+
+        before do
+          FraudMatch.update_all(resolved: true)
+          Timecop.freeze(Time.zone.local(2020, 8, 23, 12)) do
```
Comment (avinhurry) [inline spec/services/update_duplicate_matches_spec.rb:48]: I don't think we need to freeze time for this test?

File: spec/services/update_duplicate_matches_spec.rb [hunk 3]
```diff
@@ -39,6 +39,22 @@
...
+            create(:application_form, :duplicate_candidates, candidate: candidate3, submitted_at: Time.zone.now)
```
Comment (avinhurry) [inline spec/services/update_duplicate_matches_spec.rb:49]:
```suggestion
            create(:application_form, :duplicate_candidates, candidate: candidate3)
```

File: spec/services/update_duplicate_matches_spec.rb [hunk 4]
```diff
@@ -39,6 +39,22 @@
           expect(candidate2.reload.submission_blocked).to be(false)
         end
       end
+
+      context 'when the candidates have been mannually resolved and new candidate is added to the match' do
+        let(:candidate3) { create(:candidate, email_address: 'exemplar3@example.com') }
+
+        before do
+          FraudMatch.update_all(resolved: true)
```
Comment (avinhurry) [inline spec/services/update_duplicate_matches_spec.rb:47]: Is there a reason for updating all, or can we just update the first match?

File: spec/services/update_duplicate_matches_spec.rb [hunk 5]
```diff
@@ -39,6 +39,22 @@
...
+          Timecop.freeze(Time.zone.local(2020, 8, 23, 12)) do
+            create(:application_form, :duplicate_candidates, candidate: candidate3, submitted_at: Time.zone.now)
+          end
+          described_class.new.save!
+        end
+
+        it 'mark the match as unresolved' do
```
Comment (avinhurry) [inline spec/services/update_duplicate_matches_spec.rb:54]:
```suggestion
        it 'marks the match as unresolved' do
```

**Review by avinhurry — COMMENTED** [review]
Tested locally and it all works very well! Just a few minor comments inline on the tests

File: spec/services/update_duplicate_matches_spec.rb (hunk 4 above)
Comment (tomas-stefano) [reply spec/services/update_duplicate_matches_spec.rb]: I did for convenience but am happy to do only for the first one.

**Review by avinhurry — APPROVED** [review]

**Pull request merged**